### PR TITLE
feat(Mutation levels): Add mutation levels

### LINKED ISSUE #869: Feature request: Mutation strategy option
**Is your feature request related to a problem? Please describe.**
People that are new to mutation testing can be overwhelmed by the number of mutations and type of mutations that are being placed. And more experienced testers might need more detailed mutations to further improve their testing.

**Describe the solution you'd like**
I'd like to introduce a mutation strategy option that has three possible values: Light, Medium and Heavy. 

- Light excludes some mutation types to keep the report simple and the mutating fast. 
- Medium uses current mutations, which are concidered good.
- Heavy uses extra mutations like regex mutations and removes code as well. This will drasticly increase runtime. But it gives even more insight than Medium.

**Additional context**
The default would be medium. 

We have to discuss what mutations will come with each strategy. @nicojs, @Mobrockers what are your thoughts on this?

### PR DESCRIPTION
**Pull request by richardwerkman**
This PR aims to create a leveled structure for the mutations. For now it only targets mutators as a whole. But I can imagine that some Linq mutations can be `Complete` while others are `Standard`

closes #869

The mutation levels
| Mutations| Level| 
| ------------- | ------------- | 
| Arithmetic Operators | Basic|
| Block (if we ever get it done) | Basic|
| Equality Operators | Standard |
| Boolean Literals | Standard|
| Assignment statements | Standard |
| Collection initializer | Standard |
| Unary Operators | Standard |
| Update Operators | Standard |
| String Literals and Constants | Standard |
| Bitwise Operators | Standard |
| Checked Statements | Standard |
| Linq Methods | Advanced |
| Regex | Advanced |
| Advanced Linq Methods | Complete |
| Advanced Regex | Complete |

### CODE REVIEW

File: src/Stryker.CLI/Stryker.CLI/CLIOptions.cs [hunk 1]
```diff
@@ -279,6 +279,15 @@ public static class CLIOptions
             JsonKey = "test-projects"
         };
 
+        public static readonly CLIOption<string> MutationLevel = new CLIOption<string>
+        {
+            ArgumentName = "--mutation-level",
+            ArgumentShortName = "-ml",
+            ArgumentDescription = $"Specify what test projects should run on the project under test.",
```
Comment (Mobrockers) [inline src/Stryker.CLI/Stryker.CLI/CLIOptions.cs]: The description belongs to a different option

File: src/Stryker.CLI/Stryker.CLI/CLIOptions.cs [hunk 2]
```diff
@@ -279,6 +279,15 @@ public static class CLIOptions
             JsonKey = "test-projects"
         };
 
+        public static readonly CLIOption<string> MutationLevel = new CLIOption<string>
```
Comment (Mobrockers) [inline src/Stryker.CLI/Stryker.CLI/CLIOptions.cs]: I would prefer something like Mode or MutationMode

Mode > MutationMode tbh

File: src/Stryker.Core/Stryker.Core/Mutators/MutationLevel.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿namespace Stryker.Core.Mutators
+{
+    public enum MutationLevel
+    {
+        Basic,
+        Standard,
+        Advanced,
+        Complete
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Mutators/MutationLevel.cs]:
```suggestion
        Basic = 0,
        Standard = 5,
        Advanced = 10,
        Complete = 100
```

File: src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs [hunk 4]
```diff
@@ -81,7 +82,8 @@ public class StrykerOptions
             string projectName = null,
             string moduleName = null,
             string projectVersion = null,
-            IEnumerable<string> testProjects = null)
+            IEnumerable<string> testProjects = null,
+            string mutationLevel = "intermediate")
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs]:
```suggestion
            string mutationLevel = "Standard")
```

File: src/Stryker.Core/Stryker.Core/Mutators/IMutator.cs [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 using Stryker.Core.Mutants;
+using Stryker.Core.Options;
 using System.Collections.Generic;
 
 namespace Stryker.Core.Mutators
 {
     public interface IMutator
     {
-        IEnumerable<Mutation> Mutate(SyntaxNode node);
+        IEnumerable<Mutation> Mutate(SyntaxNode node, StrykerOptions mutationLevel);
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Mutators/IMutator.cs]:
```suggestion
        IEnumerable<Mutation> Mutate(SyntaxNode node, StrykerOptions options);
```

File: src/Stryker.CLI/Stryker.CLI/CLIOptions.cs (hunk 2 above)
Comment (richardwerkman) [reply src/Stryker.CLI/Stryker.CLI/CLIOptions.cs]: I disagree. Level suggests there are gradations and each level adds more mutatants. Mode says more about how mutations should be placed or something. Not about how many mutations are placed. 

I'd like to think about this like loglevel. The lower the loglevel the more logs are kept. In stryker the higher the mutationlevel the more mutations are being placed.

**Comment by richardwerkman** [comment]
@Mobrockers could you check this PR again? Especially the proposed mutators under each level. It will have a big impact, for example: the linq mutator will be disabled by default.

File: src/Stryker.CLI/Stryker.CLI/CLIOptions.cs (hunk 2 above)
Comment (richardwerkman) [reply src/Stryker.CLI/Stryker.CLI/CLIOptions.cs:356]: @Mobrockers I've heard no one else say they prefer Mode over Level so I'd say we go with Level. I'd like to merge this today so we can add complex mutators during hacktoberfest :)

**Pull request merged**